lib/tests/streamlit/runtime/metrics_util_test.py

```diff
@@ -211,3 +210,157 @@ def test_function(param1: int, param2: str, param3: float = 0.1) -> str:
         self.assertEqual(len(ctx.tracked_commands), 0)
         test_function(param1=10, param2="foobar")
         self.assertEqual(len(ctx.tracked_commands), 0)
+
+    @parameterized.expand(
+        [
+            (st._transparent_write, "magic"),
+            (st.experimental_memo.clear, "clear_memo"),
+            (st.experimental_singleton.clear, "clear_singleton"),
+            (st.session_state.__setattr__, "session_state.set_attr"),
+            (st.session_state.__setitem__, "session_state.set_item"),
+            (websocket_headers._get_websocket_headers, "_get_websocket_headers"),
+            (memo_decorator.MemoCache.write_result, "_cache_memo_object"),
+            (
+                singleton_decorator.SingletonCache.write_result,
+                "_cache_singleton_object",
+            ),
+            (caching._write_to_cache, "_cache_object"),
+        ]
+    )
+    def test_internal_api_commands(self, command: Callable, expected_name: str):
+        """Some internal functions are also tracked and should use the correct name."""
+        ctx = get_script_run_ctx()
+        ctx.reset()
+        ctx.gather_usage_stats = True
+
+        # This will always throw an exception because of missing arguments
+        # This is fine since the command still get tracked
+        with contextlib.suppress(Exception):
+            command()
+
+        self.assertGreater(
+            len(ctx.tracked_commands),
+            0,
+            f"No command tracked for {expected_name}",
+        )
+
+        # Sometimes multiple commands are executed
+        # so we check the full list of tracked commands
+        self.assertIn(
+            expected_name,
+            [tracked_commands.name for tracked_commands in ctx.tracked_commands],
+            f"Command {expected_name} was not tracked.",
+        )
+
+    def test_public_api_commands(self):
+        """All commands of the public API should be tracked with the correct name."""
+        ctx = get_script_run_ctx()
+        ctx.reset()
+        ctx.gather_usage_stats = True
+
+        # Some commands are currently not tracked for various reasons:
+        ignored_commands = [
```

Comment:
Nit: should be a `set`. (And then we can just subtract `ignored_commands` from `public_commands`)